R/utilities.r

```diff
@@ -299,6 +299,16 @@ find_args <- function(...) {
 # global data
 dummy_data <- function() data.frame(x = NA)
 
+with_seed <- function(seed, code) {
+  if (!is.null(seed)) {
+    old_seed <- get0(".Random.seed", globalenv())
```

Comment:
Good catch. The code worked by accident because *resetting* the random seed after applying the jitter also made it reproducible.